### PR TITLE
Collections excercises

### PR DESCRIPTION
**Pull request by Tadaboody**

### CODE REVIEW

File: collections/src/lib.rs [hunk 1]
```diff
@@ -32,16 +32,61 @@ fn mode<T: Eq + Copy + Hash + Ord>(it: &[T]) -> T {
...
+// Instead of printing I return a string with what should be printed.
+pub fn process_command(employees: &mut Employees, command: &str) -> Option<String> {
+    let add_command = Regex::new(r"(?i)add (?P<name>\w+) to (?P<department>\w+)").unwrap();
+    let list_command = Regex::new(r"(?i)list(?:\s(?P<department>\w+))?").unwrap();
+    if let Some(add) = add_command.captures(&command.to_ascii_lowercase()) {
+        employees
+            .entry(add.name("department").unwrap().as_str().to_owned())
+            .or_default()
+            .push(add.name("name").unwrap().as_str().to_owned());
+        None
```
Comment (selfint) [inline collections/src/lib.rs:64]: Why do you need to return None here?

Comment (selfint) [reply collections/src/lib.rs:64]: oh nvm i see now

File: collections/src/lib.rs [hunk 2]
```diff
@@ -32,16 +32,61 @@ fn mode<T: Eq + Copy + Hash + Ord>(it: &[T]) -> T {
...
+    } else if let Some(list_match) = list_command.captures(command) {
+        let format_department = |department: &str, names: &[String]| {
```
Comment (selfint) [inline collections/src/lib.rs]: why define a named function using an anonymous function?
```suggestion
        fn format_department(department: &str, names: &[String]) {
```

Comment (Tadaboody) [reply collections/src/lib.rs]: Thought I couldn't do an inner function - turns out I was wrong!

**Pull request merged**